### DISCUSSION TITLE
Number of layers with BlueBuild

### DISCUSSION THREAD

**Discussion by mikaelvz** [blue-build/.github#40] [Support] [answered]
Hi,
Does the number of layers added by the Containerfile have any impact ?

I made a recipe with a lot of instructions (by example multiple rpm-ostree module calls), It keep the recipe mode "readable" but add more layers. Should I try to optimize the recipe to limit the number of layers ?

**Comment by MrStickyPiston** [comment]
I know its been 2 months, leaving this here for people having the same questions: 

It should not have a big impact. In general readability is worth more than the tiny difference in build/update time.

**Reply by xynydev** [reply, marked as answer]
Yeah sorry for not responding earlier, but I'd like to add that you can enable build squashing if limiting the number of layers is something you'd like to do. Layers usually bring caching benefits, though, and if you order your module calls such that the least frequently changing parts are at the top it should be able to utilize layer caching in the builds and updates.
